### PR TITLE
terraform 0.12upgrade

### PR DESCRIPTION
**Pull request by stavxyz**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -33,53 +33,53 @@ terraform {
...
 
 resource "aws_dynamodb_table" "tf_backend_state_lock_table" {
-  count = "${var.dynamodb_lock_table_enabled ? 1 : 0}"
-  name           = "${var.dynamodb_lock_table_name}"
-  read_capacity  = "${var.lock_table_read_capacity}"
-  write_capacity = "${var.lock_table_write_capacity}"
-  hash_key       = "LockID"
-  stream_enabled = "${var.dynamodb_lock_table_stream_enabled}"
-  stream_view_type = "${var.dynamodb_lock_table_stream_enabled ? var.dynamodb_lock_table_stream_view_type : ""}"
+  count            = var.dynamodb_lock_table_enabled ? 1 : 0
```
Comment (richardARPANET) [inline main.tf:40]: I was getting errors like:

```

Error: Incorrect attribute value type

  on .terraform/modules/backend/main.tf line 45, in resource "aws_dynamodb_table" "tf_backend_state_lock_table":
  45:   stream_enabled   = var.dynamodb_lock_table_stream_enabled

Inappropriate value for attribute "stream_enabled": bool required.


Error: Incorrect condition type

  on .terraform/modules/backend/main.tf line 46, in resource "aws_dynamodb_table" "tf_backend_state_lock_table":
  46:   stream_view_type = var.dynamodb_lock_table_stream_enabled ? var.dynamodb_lock_table_stream_view_type : ""
    |----------------
    | var.dynamodb_lock_table_stream_enabled is 0

```
This fixed it https://github.com/richardARPANET/terraform-aws-backend/commit/8d90f0c51f71f1e9db62acb12deb73db1e301132

Comment (samstav) [reply main.tf]: I suppose all of the boolean-ish options should be changed from 0/1 to real booleans.

**Comment by stavxyz** [comment]
Hey @richardARPANET thanks for the tip. 

I suppose all of the boolean-ish options should be changed from 0/1 to real booleans.

**Comment by stavxyz** [comment]
Closing in favor of https://github.com/samstav/terraform-aws-backend/pull/18

**Pull request closed without merging**